lib/WebDriver/AbstractWebDriver.php

```diff
@@ -122,15 +124,30 @@ protected function curl($requestMethod, $command, $parameters = null, $extraOpti
 
         list($rawResult, $info) = ServiceFactory::getInstance()->getService('service.curl')->execute($requestMethod, $url, $parameters, $extraOptions);
 
+        // According to the spec:
+        // - valid responses always include: sessionId, status and value
+        //   (at least as long as 'they include a message body' - but we tested commands such as timeouts/async_script
+        //   and session/delete and they still give back a json payload)
+        // - failed commands (http 500) include those as well
+        // - invalid requests (http 4xx) are instead plaintext
+        // See https://w3c.github.io/webdriver/webdriver-spec.html for details
+
+        if ($info['http_code'] >= 400 && $info['http_code'] <= 499) {
+            throw WebDriverException::factory(Exception::CURL_EXEC, 'Webdriver http error: ' . $info['http_code'] . ', payload :' . substr($rawResult, 0, 1000)/*, $previousException*/);
```

Comment:
1000 is indeed completely arbitrary. And the unattentive user might be caught off guard (I should have really introduced at least shortening with ... if resp. length was over 1000).
It's just that http responses can be massive, and I thought having no limit was risky